### PR TITLE
Add a multisig example

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
This PR adds a multisig example (legacy sdk reference [example](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/examples/typescript/multisig_account.ts))

While working on it, I noticed some gaps in our current SDK.
- ABI parser support for a `TransactionPayloadMultiSig` without a `MultiSigTransactionPayload` 
- Offer a framework around building and interacting with multisig transactions with the SDK.

### Test Plan
- pnpm run test
- example output
```
owner1: 0xcea337802e090d160d0fa6952c4644f99af5fdbf2dd14ed205e09189872b3d00
owner2: 0x43a6e5e7e5551abae94d7362fc37bd2ca6d2d9b3bed4a4e23c9c958500b75bd4
owner3: 0x86efd860ec4680c7938226080f48b9b8465ffe9fec54b396ffa764911e28db91
Setting up a 2-of-3 multisig account...
Multisig Account Address: 0x2b253ac3ae157d46865b7126b226b4fcb75bda8f3046420da9c9a2c82f6ff75d
Signature Threshold: 2
Number of Owners: 3
Funding the multisig account...
Creating a multisig transaction to transfer coins...
Owner 1 rejects but owner 3 approves.
Owner 2 can now execute the transfer transaction as it already has 2 approvals (from owners 2 and 3).
Recipient's balance after transfer 1000000
Creating another multisig transaction using payload hash...
Owner 1 rejects but owner 3 approves.
Owner 2 can now execute the transfer with hash transaction as it already has 2 approvals (from owners 2 and 3).
Recipient's balance after transfer 2000000
Adding an owner to the multisig account...
Owner 1 rejects but owner 3 approves.
Owner 2 can now execute the adding an owner transaction as it already has 2 approvals (from owners 2 and 3).
Number of Owners: 4
Removing an owner from the multisig account...
Owner 1 rejects but owner 3 approves.
Owner 2 can now execute the removing an owner transaction as it already has 2 approvals (from owners 2 and 3).
Number of Owners: 3
Changing the signature threshold to 3-of-3...
Owner 1 rejects but owner 3 approves.
Owner 2 can now execute the change signature threshold transaction as it already has 2 approvals (from owners 2 and 3).
Signature Threshold: 3
Multisig setup and transactions complete.
```

### CODE REVIEW

File: src/transactions/types.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import {
   FeePayerRawTransaction,
   RawTransaction,
   TransactionPayloadEntryFunction,
-  TransactionPayloadMultisig,
+  TransactionPayloadMultiSig,
   TransactionPayloadScript,
```
Comment (gregnazario) [inline src/transactions/types.ts:16]: Let's make sure this recapitalization is in the changelog as breaking

File: src/transactions/transactionBuilder/transactionBuilder.ts [hunk 2]
```diff
@@ -42,11 +42,11 @@ import {
   FeePayerRawTransaction,
   MultiAgentRawTransaction,
   MultiSig,
-  MultisigTransactionPayload,
+  MultiSigTransactionPayload,
   RawTransaction,
   Script,
   TransactionPayloadEntryFunction,
-  TransactionPayloadMultisig,
+  TransactionPayloadMultiSig,
```
Comment (gregnazario) [inline src/transactions/transactionBuilder/transactionBuilder.ts:49]: Similarly

File: src/transactions/instances/transactionPayload.ts [hunk 3]
```diff
@@ -404,9 +405,10 @@ export class MultisigTransactionPayload {
...
   }
 
-  static deserialize(deserializer: Deserializer): MultisigTransactionPayload {
-    // This is the enum value indicating which type of payload the multi-sig transaction contains.
+  static deserialize(deserializer: Deserializer): MultiSigTransactionPayload {
+    // TODO: Support other types of payload beside EntryFunction.
+    // This is the enum value indicating which type of payload the multisig tx contains.
     deserializer.deserializeUleb128AsU32();
-    return new MultisigTransactionPayload(EntryFunction.deserialize(deserializer));
+    return new MultiSigTransactionPayload(EntryFunction.deserialize(deserializer));
```
Comment (gregnazario) [inline src/transactions/instances/transactionPayload.ts:412]: This has to be done on the Aptos Move framework level.  There is no way to support non-entry functions afaik at the moment.  cc @movekevin

Comment (0xmaayan) [reply src/transactions/instances/transactionPayload.ts:412]: This is a legacy comment https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/aptos_types/transaction.ts#L228

**Pull request merged**